assets/app/features/upgrade/controllers/upgrade7-upgrade-admin.controller.spec.js

```diff
@@ -0,0 +1,161 @@
+                    $rootScope.$digest();
+                });
+
+                it('should set running to true', function () {
+                    assert.isTrue(controller.adminUpgrade.running);
+                });
+
+                it('should call checkAdminUpgrade() to start polling', function () {
+                    expect(controller.adminUpgrade.checkAdminUpgrade).toHaveBeenCalledTimes(1);
+                });
+            });
+
+            describe('when starting upgrade failed', function () {
+                beforeEach(function () {
+                    spyOn(controller.adminUpgrade, 'checkAdminUpgrade');
+
+                    bard.mockService(upgradeUpgradeAdminFactory, {
+                        getAdminUpgrade: $q.reject(errorResponse)
+                    });
+                    controller.adminUpgrade.beginAdminUpgrade();
+                    $rootScope.$digest();
+                });
+
+                it('should leave running at false', function () {
+                    assert.isFalse(controller.adminUpgrade.running);
+                });
+
+                it('should not call checkAdminUpgrade()', function () {
+                    expect(controller.adminUpgrade.checkAdminUpgrade).not.toHaveBeenCalled();
+                });
+
+                it('should expose the errors through adminUpgrade.errors object', function () {
+                    expect(controller.adminUpgrade.errors).toEqual(errorList);
+                });
+            });
+
+        });
+
+        describe('checkAdminUpgrade function', function () {
+            it('should be defined', function () {
+                should.exist(controller.adminUpgrade.checkAdminUpgrade);
+            });
+
+            describe('when got upgrade status from api successfully', function () {
+                describe('when received status is completed', function () {
+                    it('should enable "Next button"', function () {});
+                    it('should set running attribute of adminUpgrade model to false', function () {});
+                    it('should set completed flag set to true', function () {});
+                });
+
+                describe('when received status is not completed', function () {
+                    it('should keep "Next button" disabled', function () {});
+                    it('should keep running flag set to true', function () {});
+                    it('should keep completed flag set to false', function () {});
+                    it('should schedule another check', function () {});
+                });
+            });
+
+            describe('when got error from api', function () {
+                it('should expose the errors through adminUpgrade.errors object', function () {
+//                    expect(controller.adminUpgrade.errors).toEqual(errorList);
+                });
+            });
+        });
+    });
+
+/*    describe('cancelUpgrade function', function () {
```

Comment:
Same as `Next Step` on https://github.com/crowbar/crowbar-angular/pull/23/files#r77822851
CancelUpgrade functionality should be part of the Upgrade7 controller, so I think this section could be removed